### PR TITLE
Add Mockito test for ArtifactListPresenterImpl

### PR DESCRIPTION
**Pull request by yurloc**
Includes changes necessary to avoid GwtMockito and some minor refactoring.

I have rebuilt both `kie-wb-distribution-wars` and `kie-drools-wb-distribution-wars` and tested changes in Firefox 40 and IE 11.

### CODE REVIEW

File: guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListPresenterImpl.java [hunk 1]
```diff
@@ -36,58 +35,67 @@
...
+    @Override
+    public ArtifactListView getView() {
+        return view;
     }
 
     @Override
     public void refresh() {
-        // We couldn't attach the Display to the DataProvider in the @PostConstruct method
-        // as this initializes the Display with data using AsyncDataProvider.onRangeChanged()
-        // at which time the View may not have been initialised fully.
```
Comment (yurloc) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListPresenterImpl.java]: @manstis Do you remember what exactly did you mean by *the view may have not been initialised fully*? IMO the view instance is already fully initialized when injected into presenter constructor. The only thing I can imagine is that an additional column is [conditionally added](https://github.com/droolsjbpm/guvnor/blob/master/guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/editor/MavenRepositoryPagedJarTable.java#L56-L78) in parent component's `@PostConstruct` but I have tested the change manually and everything works as expected.

File: guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListViewImpl.java [hunk 2]
```diff
@@ -46,11 +43,6 @@
 
     protected ArtifactListPresenter presenter;
 
-    @Inject
-    protected Caller<M2RepoService> m2RepoService;
```
Comment (yurloc) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListViewImpl.java]: Interactions with backend services are handled by presenter.

File: guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListViewImpl.java [hunk 3]
```diff
@@ -46,11 +43,6 @@
...
-
-    protected String currentFilter = "";
```
Comment (yurloc) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListViewImpl.java]: View doesn't need to know about the filter. This responsibility has been moved to data provider.

File: guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListPresenterImpl.java (hunk 1 above)
Comment (manstis) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListPresenterImpl.java]: @yurloc hehe. as if I knew ;)... I recall it has something to do with the re-use of the ```ArtifactListPresenter``` and related ```ArtifactListView``` in the Project Editor's "Add dependency from Repository" view.. see [here](https://github.com/droolsjbpm/kie-wb-common/blob/master/kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/forms/DependencyListWidget.java#L77).

**Comment by manstis** [comment]
@yurloc Hi Jiri, if you can confirm the ```ArtifactListView``` works OK in both the "Guvnor M2 Repository" screen and the "Project Editor->Add Dependency from Repository" I'll be happy to merge this PR.. IDK if you've checked the Project Editor's use....

**Comment by yurloc** [comment]
@manstis Yes, I have tested the dependency dialog in Project Editor and couldn't find anything wrong.

**Comment by manstis** [comment]
*guvnor*
(master) http://github.com/droolsjbpm/guvnor/commit/f7b855beb

(6.3.x) http://github.com/droolsjbpm/guvnor/commit/65e5f9d96

**Comment by manstis** [comment]
Thanks @yurloc

**Pull request closed without merging**